source/agora/node/Registry.d

```diff
@@ -680,6 +687,171 @@ private struct ZoneData
     /// The content of the zone
     public TypedPayload[PublicKey] map;
 
+    /// Query for registry count interface
+    private string query_count;
+
+    /// Query for getting all registries
+    private string query_registry_get;
+
+    /// Query for getting payload
+    private string query_payload;
+
+    /// Query for adding registry signature table
+    private string query_signature_add;
+
+    /// Query for adding registry to addresses table
+    private string query_addresses_add;
+
+    /// Database to store data
+    private ManagedDatabase db;
+
+    /***************************************************************************
+
+         Params:
+           type_table_name = Registry table for table name
+           cache_db = Database instance
+
+    ***************************************************************************/
+
+    public this (string type_table_name, ManagedDatabase cache_db)
+    {
+        this.db = cache_db;
+
+        this.query_count = format("SELECT COUNT(*) FROM registry_%s_signature", type_table_name);
```

Comment:
This works, but templated `ZoneData` could be a better idea. That would let you inline these.

`"SELECT COUNT(*) FROM registry_" ~ TemplateTableName ~ "_signature"` would work perfectly fine with no runtime allocations IIRC.

Reply:
It's good idea, but `ZoneData` will be initiated two times at most in a Registry and allocations will only made during initialization. Last commit implements the suggestion and I think it results in cleaner `ZoneData` attributes.